lib/core/src/Cardano/Pool/DB/Model.hs

```diff
@@ -416,6 +422,15 @@ mRemoveRetiredPools epoch = do
     mRemovePools (view #poolId <$> certificates)
     pure certificates
 
+mPutHeader :: BlockHeader -> ModelOp ()
+mPutHeader header = modify #blockHeaders (header :)
+
+mListHeaders :: Int -> ModelOp [BlockHeader]
+mListHeaders k
+    | k > 0 = reverse . take k <$> get #blockHeaders
+    | otherwise = reverse <$> get #blockHeaders
```

Comment:
Hi @hasufell, is there a good reason for this special case (where `k == 0`) to exist?

If not, I would be inclined to remove it.

It also seems that this behaviour is different from the SQLite implementation, which just returns 0 rows if `k` is zero. (Though please correct me if I'm wrong here!)

Reply:
> It also seems that this behaviour is different from the SQLite implementation, which just returns 0 rows if k is zero. (Though please correct me if I'm wrong here!)

I returns all elements if k is 0.

Reply:
> > It also seems that this behaviour is different from the SQLite implementation, which just returns 0 rows if k is zero. (Though please correct me if I'm wrong here!)
> 
> I returns all elements if k is 0.

Is this documented anywhere?

I couldn't find any mention of this in [the Haddock documentation for `LimitTo`](https://hackage.haskell.org/package/persistent-2.10.5.2/docs/Database-Persist-Types.html#t:SelectOpt).

If this is the case, then we should definitely add a comment in the interface for `DBLayer.listHeaders`, to let callers know this is what they should expect.